bson/_cbsonmodule.c

```diff
@@ -455,25 +475,30 @@ static long _type_marker(PyObject* object) {
  */
 int convert_codec_options(PyObject* options_obj, void* p) {
     codec_options_t* options = (codec_options_t*)p;
-    long type_marker;
     options->unicode_decode_error_handler = NULL;
-    if (!PyArg_ParseTuple(options_obj, "ObbzO",
+    PyObject* type_registry_obj = NULL;
+    long type_marker;
+
+    if (!PyArg_ParseTuple(options_obj, "ObbzOO",
                           &options->document_class,
                           &options->tz_aware,
                           &options->uuid_rep,
                           &options->unicode_decode_error_handler,
-                          &options->tzinfo)) {
+                          &options->tzinfo,
+                          &type_registry_obj))
         return 0;
-    }
+
+    convert_type_registry(type_registry_obj, &options->type_registry);
 
     type_marker = _type_marker(options->document_class);
     if (type_marker < 0) return 0;
```

Comment:
There's a potential leak here because `convert_type_registry` increments the reference count of `type_registry_obj` and we can return failure here without a corresponding decref.

I think the simplest fix is to call `convert_type_registry` *after*  `if (type_marker < 0) return 0;`.